### ISSUE TITLE
Feature Request: Please also support Ollama models

### ISSUE THREAD

**Issue by kroll-software** [microsoft/TinyTroupe#17]
I'd like to run it locally.

Thanks in advance,

**Comment by paulosalem** [comment]
Thanks for the suggestion! I agree this would be useful. However, support for new LLMs is not a priority at the moment, because there's still a lot of core work to be done (leveraging current LLM endpoints), and we are optimizing the system to work as well as possible with OpenAI models. But I'll leave the issue here as a feature request to keep it in our radar for the time being.

**Comment by Katzukum** [comment]
> It should be easy to implement with python-dotenv and an .env file with just two lines
> 
> OPENAI_BASE_URL=http://localhost:11434/v1 OPENAI_API_KEY=fake-key
> 
> Ollama provides an OpenAI compatible API on the local computer, the calls are simply directed there.
> 
> Best regards,

its a quick fix you can edit, I did the following 

Added the following to the config file. 
END_POINT = http://127.0.0.1:1234/v1/
API_KEY = lm-studio

and Updated openai_utils.py in the site packages or the original file before the install with 

self.client = OpenAI(api_key=config["OpenAI"]["API_KEY"], base_url=config["OpenAI"]["END_POINT"])#os.getenv("OPENAI_API_KEY"))

So far this has worked for the base examples, I am still testing a more complex example that uses all the features.

**Comment by d8rt8v** [comment]
Thx @Katzukum 

It seems to me, when working with local LLMs, you might want to comment out `def _count_tokens` since it’s no longer necessary and can interfere with the models in LM Studio (though llama 3.2 works fine, but had a problem with `aya-expanse`). 

Also, if you’re using a resource-intensive model, consider setting `TIMEOUT=100` or more in `config.ini`!

**Comment by paulosalem** [comment]
Update: we actually have an ongoing PR proposal that would enable this: https://github.com/microsoft/TinyTroupe/pull/47

As these are complex changes, it might take some time to review, but the fact that someone already did a lot of heavy lifting here helps a lot, and I'll try to review it with care.

**Comment by easis** [comment]
See https://github.com/microsoft/TinyTroupe/discussions/99

TL;DR

```python
import os

os.environ["OPENAI_API_KEY"] = "ollama"
os.environ["OPENAI_BASE_URL"] = "http://localhost:11434/v1"
```

**Comment by kroll-software** [comment]
great to hear, thank you, I'll try it out.
